remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java

```diff
@@ -124,9 +126,19 @@ public int size()
             {
 		        List<Pair<String, NodeRef>> tagNodeRefs = taggingService.addTags(nodeRef, tagValues);
 		        List<Tag> ret = new ArrayList<Tag>(tags.size());
+				tagsByCount = taggingService.findTaggedNodesAndCountByTagName(nodeRef.getStoreRef());
+				if (tagsByCount != null)
+				{
+					for (Pair<String, Integer> tagByCountElem : tagsByCount)
+					{
+						tagsByCountMap.put(tagByCountElem.getFirst(), tagByCountElem.getSecond());
+					}
+				}
 		        for(Pair<String, NodeRef> pair : tagNodeRefs)
 		        {
-		        	ret.add(new Tag(pair.getSecond(), pair.getFirst()));
+					Tag createdTag=new Tag(pair.getSecond(), pair.getFirst());
+					createdTag.setCount(Optional.ofNullable(tagsByCountMap.get(createdTag.getTag())).orElse(0));
```

Comment:
As discussed this will usually be out by one, since the tagged node has not been indexed yet. It might be better to always assume the count will be one more than whatever the index says?